### PR TITLE
f32/f64: add MulAdd (dst += a*b) fused multiply-accumulate

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Adds `MulAdd(dst, a, b)` to the `f32` and `f64` packages, computing `dst[i] += a[i]*b[i]` (a fused multiply-accumulate in place).

It is implemented as a thin wrapper over the existing `FMA` kernel: `MulAdd(dst, a, b)` calls the internal `fma` kernel with `c` aliased to `dst`, i.e. it is exactly `FMA(dst, a, b, dst)`. This reuses the already dispatched and contract-tested SIMD kernels (AVX-512, AVX+FMA, SSE2 on amd64; NEON on arm64; pure-Go fallback) with no new assembly, and inherits FMA's tolerance-stable numeric contract: the multiply-add is fused with a single rounding on the hardware-FMA tiers (AVX+FMA, AVX-512, NEON), and is a separate multiply then add on SSE2 and on the f32 pure-Go path. The f64 pure-Go fallback fuses via `math.FMA`. Results are tolerance-stable across tiers, not bit-identical.

Motivation: a common STFT/overlap-add pattern windows a frame and accumulates it into an output buffer with a separate `Mul` followed by `Add`, making two passes over the frame per hop. `MulAdd` fuses that into one pass and removes the intermediate write-back. Motivated by tphakala/go-audio-dsp#27.

`dst` is a read-modify-write accumulator and, following the package default, may exactly overlay `a` and/or `b`; a shifted overlay is undefined. No `Unsafe` variant is added, matching `FMA`.

## Tests

- Correctness across every dispatch tier: a bit-exact invariant against `FMA(dst, a, b, dst)` and parity against a scalar reference within tolerance.
- Empty and mismatched-length edge cases (clamp to the shortest slice).
- Allocation-free on the separate-destination path and on the in-place overlays.
- A bespoke read-modify-write aliasing sweep (`dst==a`, `dst==b`, `dst==a==b`) with zero-allocation checks, mirroring the existing `AddScaled` sweep.
- A fuzz target against the scalar reference within tolerance.
- Godoc examples and a README operations-table row.

Verified on amd64 (AVX, AVX-without-FMA, SSE2, and pure-Go tiers) and on arm64 NEON (Raspberry Pi 5, Cortex-A76). `golangci-lint run ./...` is clean.

### CODE REVIEW

File: f32/f32.go [hunk 1]
```diff
@@ -294,6 +294,24 @@ func FMA(dst, a, b, c []float32) {
 	fma32(dst[:n], a[:n], b[:n], c[:n])
 }
 
+// MulAdd computes the fused multiply-accumulate dst[i] += a[i] * b[i].
+//
+// It is equivalent to FMA(dst, a, b, dst) and shares FMA's numerical contract:
+// on CPU tiers with hardware FMA (AVX+FMA, AVX-512, NEON) the multiply-add is
+// fused with a single rounding; on the SSE2 and pure-Go amd64 paths it is a
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32.go]: Good catch. Fixed in 689c2d5: reworded to refer to the pure-Go fallback generally rather than amd64, since the f32 pure-Go fma path also runs on non-NEON arm64, sub-minimum slices, and other architectures.
